tests/test_dataloaders.py

```diff
@@ -301,6 +301,11 @@ async def idx(keys: List[int]) -> List[Tuple[int, int]]:
 
     loader = DataLoader(load_fn=idx, cache=False)
 
+    try:
+        loader.clean(1)  # no effect on non-cached values
+    except Exception as e:
+        raise AssertionError("clean non-cached values does not raise KeyError")
```

Comment:
I'm not sure what your intention is here. Normally a test will implement desired behaviour, and if it works without errors then (that part of) the test passes. So we could simplify to this ...

```suggestion
    loader.clear(1)  # no effect on non-cached values
```

...which is similar to the other calls below to `loader.clear()` / `loader.clear_many()` / etc

Reply:
You should also add this to the `test_clear()` test function above, so that we can exercise the `.clear()` functionality with an underlying cache

See https://github.com/strawberry-graphql/strawberry/pull/3569/files#diff-9e2920b06c9fb1a3a76b9f675328e820421756fa44783ebdd45ca2c7a047c8a7R266-R279